hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp

```diff
@@ -292,7 +292,7 @@ namespace hpx { namespace threads { namespace detail
                          << " failed with: " << e.what();
 
             // trigger the barrier
-            pool_threads -= thread_num;
+            pool_threads -= (thread_num + 1);
```

Comment:
Why did you make this change?

Reply:
Without this change the loop below tries to wait for one too many thread. `thread_num` gets incremented only after the thread has successfully started, so any exception thrown during thread startup will see `thread_num` not reflecting the current (failed) thread.

Reply:
Shouldn't it be rather `--pool_threads` then? `thread_num + 1` itself might decrease the value too much, no?

Reply:
what's the difference between
```
    pool_threads -= (thread_num + 1); 
```
and
```
    --pool_threads;
    pool_threads -= thread_num; 
```
?

Reply:
I meant just `--pool_threads`, not both.

Reply:
That will not work.

Reply:
Ok, missed that the catch block is covering the loop as well.

Reply:
I take that as an approval of this PR...